arch/common/CMakeLists.txt

```diff
@@ -34,7 +34,9 @@ zephyr_linker_sources_ifdef(CONFIG_NOCACHE_MEMORY
   nocache.ld
 )
 
-# Only ARM, X86 and RISCV use TEXT_SECTION_OFFSET
-if (DEFINED CONFIG_ARM OR DEFINED CONFIG_X86 OR DEFINED CONFIG_RISCV)
+# Only ARM, X86 and RISCV use TEXT_SECTION_OFFSET.
+if (DEFINED CONFIG_X86 OR DEFINED CONFIG_RISCV)
   zephyr_linker_sources(TEXT_START SORT_KEY 0x0 text_section_offset.ld)
+elseif (DEFINED CONFIG_ARM)
+  zephyr_linker_sources(ROM_START  SORT_KEY 0x0 text_section_offset_section.ld)
```

Comment:
This thing needs to be re-factored properly, IMO.
- we need to comment inline why the ARM text section offset is different from RISC-V, X86
- we need to try to combine have a single file for all these ARCHes to implement the section offset.

Reply:
Do you want the comments here in the CMakeLists.txt file?

Reply:
Ideally, I would like _not_ to have the if, else if in CmakeLists, and apply any required conditioning inside the single text_Section_offset.ld

Reply:
update: you could keep the if, else here, but is it possible to have the same file linked, i.e. text_Section_offset.ld?

Reply:
Merged into one ifdefed file